File: trabajo 2/recorrer_arr.py
arr = [0] * 5
z = 1
res = False

print("ingrese su opcion")
print("[1] recorrer y agregar")
print("[2] recorrer, agregar y encontrar")
op = int(input("> "))

if op == 1:
    print("recorrer arreglo\n")
    for i in range(5):
        arr[i] = int(input("ingresa un numero: "))

    print("este es su arreglo:")
    print(",".join(map(str, arr)))

    while z > 0:
        x = int(input("ingrese en donde quiere sobrescribir un valor [del 0 al 4]: "))
        cab = int(input("ingrese el nuevo numero: "))
        arr[x] = cab
        z = int(input("quieres seguir cambiando el arreglo? si = 1, no = 0: "))

    print("asi se ve su arreglo ahora:")
    print(",".join(map(str, arr)))

elif op == 2:
    print("recorrer arreglo y buscar\n")
    for i in range(5):
        arr[i] = int(input("ingresa un numero: "))

    cab = int(input("ingrese cual numero quieres buscar de tu arreglo: "))
    x = 0

    while x < 5 and not res:
        if cab != arr[x]:
            print("en este cajon no esta..")
            x += 1
        else:
            print(f"hey.. lo encontre aqui: {x}")
            res = True

    if not res:
        print("el numero no se encuentra en el arreglo")

File: trabajo 2/recorrer_arr.js
const readline = require("readline-sync");

let arr = new Array(5);
let z = 1;
let res = false;

console.log("ingrese su opcion");
console.log("[1] recorrer y agregar");
console.log("[2] recorrer, agregar y encontrar");
let op = parseInt(readline.question("> "));

switch (op) {
    case 1:
        console.log("recorrer arreglo\n");
        for (let i = 0; i < 5; i++) {
            arr[i] = parseInt(readline.question("ingresa un numero: "));
        }

        console.log("este es su arreglo:");
        console.log(arr.join(","));

        while (z > 0) {
            let x = parseInt(readline.question("ingrese en donde quiere sobrescribir un valor [del 0 al 4]: "));
            let cab = parseInt(readline.question("ingrese el nuevo numero: "));

            arr[x] = cab;
            z = parseInt(readline.question("quieres seguir cambiando el arreglo? si = 1, no = 0: "));
        }

        console.log("asi se ve su arreglo ahora:");
        console.log(arr.join(","));
        break;

    case 2:
        console.log("recorrer arreglo y buscar\n");
        for (let i = 0; i < 5; i++) {
            arr[i] = parseInt(readline.question("ingresa un numero: "));
        }

        let cab = parseInt(readline.question("ingrese cual numero quieres buscar de tu arreglo: "));
        let x = 0;

        while (x < 5 && !res) {
            if (cab !== arr[x]) {
                console.log("en este cajon no esta..");
                x++;
            } else {
                console.log("hey.. lo encontre aqui: " + x);
                res = true;
            }
        }

        if (!res) {
            console.log("el numero no se encuentra en el arreglo");
        }
        break;
}